general-concepts/slotting/text.xml

```diff
@@ -60,9 +60,110 @@ automatically rebuilt</uri> when the subslot of a runtime dependency changes.
+</p>
+
+<note>
+Care must be taken when using sub-slots in a library ebuild for the first time.
+Adding sub-slots will trigger rebuilds for all the packages that already use
+slot-operator dependencies (e.g. switching from SLOT="0" to SLOT="0/14" in
+<c>media-libs/libpng</c> and package <c>foo</c> depends on <c>libpng:0=</c>).
+Therefore, it's best if you start using sub-slots in the library when the
+existing library interface changes.
+</note>
+</body>
+
+<subsection>
+<title>ABI breakage</title>
+<body>
+
+<p>
+There are two ways a library can break compatibility with its consumers.
+</p>
+
+<dl>
+  <dd>
+    ABI (Application Binary Interface): this affects binaries built before the
+    change. Applications linked against a library pre-ABI break may not work
+    correctly after the break. These changes are related to internal structure,
+    such as the size of a <c>struct</c> or the type of an argument (e.g.
+    integer width). Fixing this requires a <e>rebuild</e> of all consumers.
+  </dd>
+  <dd>
+    API (Application Programming Interface): this affects consumers at compile
+    time and usually occurs when a library has deprecated and then removed a
+    function. Fixing this requires a <e>code change</e> in consumers.
+  </dd>
+</dl>
+
+<p>
+Note that subslots are not used exclusively for this purpose. While they form
+the majority of uses in the Gentoo tree, subslots may have a meaning that
+is completely divorced from SONAMEs or ABI breakage. Check
+the usage in the relevant packages before using a subslot operator!
+</p>
+
+<p>
+When made aware of ABI breakage, change the subslot. Note that
+the subslot does not have to strictly be the SONAME and therefore could be
+an arbitrary string (following naming rules).
+</p>
+
+<p>
+Be aware that some upstreams may make releases without verifying if binary
+compatibility has been broken in a minor release. You should check using
+tools like <c>dev-util/libabigail</c> or <e>ABI Laboratory</e> (available
+in Gentoo as <c>dev-util/abi-compliance-checker</c> if you prefer the non-web
+version).
+</p>
+
+<p>
+Generally, consumers <e>which link against</e> a library possessing a subslot
+that <e>represents SONAME or binary compatibility</e> should subscribe to
+it (request to be rebuilt when the subslot changes) with <c>:=</c>. Also, see
+the QA Policy Guide for information on
+<uri link="https://projects.gentoo.org/qa/policy-guide/dependencies.html#proactive-use-of-slot-operators">
+proactively subscribing to subslots</uri> before they are defined.
+</p>
+</body>
+
+<subsubsection>
+<title>General naming of a sub-slot</title>
+<body>
+
+<p>
+As a simple rule of thumb, the SONAME is usually a function of the library's
+linking filename <c>libfoo.so</c> and its <b>first version component</b>.
+The remaining version components are useful for ensuring a monotonic upgrade
+path of consumers, but aren't incorporated into the library's SONAME, which in
+this case would be <c>libfoo.so.1</c>.
```

Comment:
Since there are few exceptions which have more version components in their SONAMEs (e.g. `dev-libs/boost`), maybe devmanual should say to also actually verify SONAMEs using one of known commands?
```
readelf -d ${library} | grep SONAME
scanelf -S ${libraries}
scanelf -qF "#F%S" ${library}
scanelf -qF "%F %S" ${libraries}
scanelf -qyF "%F %S" ${libraries}
```

Reply:
Can you file a new PR for this please?